internal/core/application/wallet_service.go

```diff
@@ -845,3 +868,167 @@ func getBalancesByAsset(unspents []explorer.Utxo) map[string]BalanceInfo {
 	}
 	return balances
 }
+
+func fetchUnspents(
```

Comment:
fetchUnspent actually:
1. fetches utxos
2. converts utxos to unspnts
3. stors unspents

Reply:
yes, just like you do in your unnamed function within RelaodUtxo. fetchUnspents expects a list of AddressInfo, so you would first retrieve all addresses info from the vault and then pass down the list to fetchUnspents.

Reply:
i was thinking maybe to rename it since it is doing 3 things :)

Reply:
just a suggestion